File: app/src/main/java/com/einthefrog/historyquiz/ui/util/ColorUtil.java
package com.einthefrog.historyquiz.ui.util;

import android.content.Context;
import android.graphics.Color;

import androidx.annotation.ColorInt;
import androidx.core.content.ContextCompat;

public class ColorUtil {
    @ColorInt
    public static int colorWithAlphaFromResource(Context context, int resource, float alpha) {
        @ColorInt int color = colorFromResource(context, resource);
        return colorWithAlpha(color, alpha);
    }

    @ColorInt
    public static int colorFromResource(Context context, int resource) {
        @ColorInt int color = ContextCompat.getColor(context, resource);
        return color;
    }

    @ColorInt
    public static int colorWithAlpha(@ColorInt int color, float alphaFloat) {
        int alpha = (int) (alphaFloat * 255);
        int red = Color.red(color);
        int green = Color.green(color);
        int blue = Color.blue(color);
        return Color.argb(alpha, red, green, blue);
    }
}

File: app/src/main/java/com/einthefrog/historyquiz/ui/util/ColorUtil.kt
package com.einthefrog.historyquiz.ui.util

import android.content.Context
import android.graphics.Color
import androidx.annotation.ColorInt
import androidx.core.content.ContextCompat

object ColorUtil {
    private const val defaultColor = Color.BLACK

    @ColorInt
    fun colorWithAlphaFromResource(context: Context?, resource: Int, alpha: Float): Int {
        @ColorInt val color = colorFromResource(context, resource)
        return colorWithAlpha(color, alpha)
    }

    @ColorInt
    fun colorFromResource(context: Context?, resource: Int): Int {
        if (context == null) return defaultColor
        return ContextCompat.getColor(context, resource)
    }

    @ColorInt
    fun colorWithAlpha(@ColorInt color: Int, alphaFloat: Float): Int {
        val alpha = (alphaFloat * 255).toInt()
        val red = Color.red(color)
        val green = Color.green(color)
        val blue = Color.blue(color)
        return Color.argb(alpha, red, green, blue)
    }
}
